pkg/annotations/dynamic-network-status.go

```diff
@@ -0,0 +1,79 @@
+package annotations
+
+import (
+	"encoding/json"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	nettypes "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	multusapi "gopkg.in/k8snetworkplumbingwg/multus-cni.v3/pkg/server/api"
+)
+
+func AddDynamicIfaceToStatus(currentPod *corev1.Pod, networkSelectionElement *nettypes.NetworkSelectionElement, response *multusapi.Response) (string, error) {
+	currentIfaceStatus, err := podDynamicNetworkStatus(currentPod)
+	if err != nil {
+		return "", err
+	}
+
+	newIfaceStatus := nettypes.NetworkStatus{
+		Name: fmt.Sprintf("%s/%s", networkSelectionElement.Namespace, networkSelectionElement.Name),
+	}
+
+	if response != nil && response.Result != nil {
+		for _, iface := range response.Result.Interfaces {
+			newIfaceStatus.Interface = iface.Name
+			newIfaceStatus.Mac = iface.Mac
+		}
+
+		var ips []string
+		for _, ifaceIPConfig := range response.Result.IPs {
+			ips = append(ips, ifaceIPConfig.Address.String())
+		}
+		if len(ips) > 0 {
+			newIfaceStatus.IPs = ips
+		}
+	}
+
+	newIfaceString, err := json.Marshal(append(currentIfaceStatus, newIfaceStatus))
+	if err != nil {
+		return "", fmt.Errorf("failed to marshall the dynamic networks status")
+	}
+	return string(newIfaceString), nil
+}
+
+func DeleteDynamicIfaceFromStatus(currentPod *corev1.Pod, netName string, ifaceName string) (string, error) {
+	currentIfaceStatus, err := podDynamicNetworkStatus(currentPod)
+	if err != nil {
+		return "", err
+	}
+
+	var newIfaceStatus []nettypes.NetworkStatus
+	newIfaceStatus = make([]nettypes.NetworkStatus, 0)
+	for i := range currentIfaceStatus {
+		if currentIfaceStatus[i].Name == netName && currentIfaceStatus[i].Interface == ifaceName {
+			continue
+		}
+		newIfaceStatus = append(newIfaceStatus, currentIfaceStatus[i])
+	}
+
+	newIfaceString, err := json.Marshal(newIfaceStatus)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshall the dynamic networks status")
+	}
+	return string(newIfaceString), nil
+}
+
+func podDynamicNetworkStatus(currentPod *corev1.Pod) ([]nettypes.NetworkStatus, error) {
+	var currentIfaceStatus []nettypes.NetworkStatus
+	if currentIfaceStatusString, wasFound := currentPod.Annotations[nettypes.NetworkStatusAnnot]; wasFound {
```

Comment:
If there is no Annotations this can fail, check first with `Annotation == nil`

Reply:
You'll always have annotations - otherwise the controller will not pick up this event.